doc/i2c_bus_lock_prevention/i2c_bus_lock_prevention_hld.md

```diff
@@ -0,0 +1,425 @@
+Given that I2C Dev 6 pulls the I2C signal line.
+Here are the brief illustrated steps for the key events and actions.
+    1. I2C Dev 6 pulls the I2C signal line permanently.
+    2. CPU detects the I2C bus lock, it takes the action to reset I2C Mux1 and I2C Mux2.
+    3. All channels are closed on I2C Mux1 and I2C Mux2. The faulty device I2C Dev6 will not impact the signal lines before I2C Mux1. So CPU is able to operate on the bus now.
+    4. I2C Dev 6 and 7 need to put into the isolation list.
+
+#### 7.2. The Method to Detect I2C Bus Lock
+
+On a specific hardware model, choose three representative i2c devices to poll.
+
+The criteria to choose the representative i2c devices are shown below.
+* In order to minimize the impact to the system performance, the chosen i2c device is seldom visited by the platform driver daemons.
+* The polling interval shall be defined to a value that will not impact the system performance severely.
+
+When i2c bus lock happens, the error realized by the i2c driver is timeout error. There are two possible causes listed below for the timeout error.
+    1. The i2c slave device is dead so it does not response to the i2c command issued from the i2c controller(i.e. master device)
+    2. The i2c bus is locked. So the i2c slave device cannot receive the i2c command issued from the i2c controller.
+
+In order to detect the condition 2 correctly, it is required to poll three i2c devices to confirm that the i2c bus is REALLY locked. Given that only one i2c device is polled and it happens the device is broken and not respond to the i2c controller(i.e. condition 1 listed above). It gets the false alarm because only the i2c slave device chosen to poll is broken and not response. Actually, the i2c bus is still good and the i2c controller is able to issue i2c transactions to other i2c devices on the bus.
+
+The i2c bus is determined to be in lock state when all of the chosen i2c devices cannot be accessed correctly.
+
+#### 7.3. The Method to Recover I2C Bus Lock State
+
+When the i2c bus lock happens, the CPU triggers the reset signal to i2c mux devices to close all of the channels on these mux devices.
+
+#### 7.4. General Pseudo Design
+
+* Monitoring i2c bus health status
+    - Log when i2c bus health status is changed to bad or changed to good.
+
+* Handling i2c bus lock
+    - Reset I2C muxe devices to recover the lock state.
+    - It is known that the thermal sensors readings are accessed through the i2c bus. The fan speed control is also done through i2c bus. When i2c bus is locked, the daemon to enforce thermal policy is not working anymore. To alleviate the risk of the hazard of overheating, this design invokes the special script to set all of the fans on the switch device to full speed as long as the i2c bus health is back to normal. Given that the i2c operations to set fan speed will not be affected by the faulty i2c device, all of the fan speed can be set as full speed running.

```

Comment:
Special handling is needed to determine whether the i2c device is a temperature sensor or not. 

Please clarify if there is special handling for other types of devices. How about the fans themselves? Transceivers, PSUs, etc.? If not, please indicate as such.